src/output.rs

```diff
@@ -17,10 +18,14 @@ pub enum Argument {
 }
 
 impl FilterPrototype {
-    pub fn new(name: &str, arguments: Vec<Argument>) -> FilterPrototype {
+    pub fn new(name: &str,
+               arguments: Vec<Argument>,
+               named_arguments: Vec<(String, Argument)>)
+               -> FilterPrototype {
         FilterPrototype {
             name: name.to_owned(),
             arguments: arguments,
+            named_arguments: named_arguments,
```

Comment:
So this change will allow parsing of `named_arguments` but we don't seem to apply them (see `Output::apply_filters`).
- What is the plan for actually applying the named arguments?
- What exactly are the semantics for named arguments that we need to support, like mixing positional and named?  How can we make this easy for filter writers?  How do ensure performance?